### DISCUSSION TITLE
DBAPIError "Can't open lib ..." with SQLAlchemy versions >=1.4.16

### DISCUSSION THREAD

**Discussion by seth-p** [sqlalchemy/sqlalchemy#6602] [Usage Questions] [answered]
[I'm afraid I can't email from my work machine on which I'm encountering this issue, so can't send a full stack trace, or reproducible code.]
Using Python 3.9.4 on linux, my program runs fine using SQLAlchemy versions <=1.4.15, but with versions >=1.4.16 I get the following error message:
```
sqlalchemy.exc.DBAPIError: (pyodbc.Error) ('01000', "[01000] [unixODBC][Driver Manager]Can't open lib 'ODBC Driver 17 for SQL Server?autocommit=True' : file not found (0) (SQLDriverConnect)")
```
Nothing else changes other than the SQLAlchemy version (installed via `conda install sqlalchemy=1.4.xx`).
The connection url is of the form `'mssql+pyodbc://@HOST/DATABASE?driver=ODBC+Driver+17+for+SQL+Server?autocommit=True'`.

The error occurs in `MetaData.reflect()`:
```
metadata = sa.MetaData()
metadata.reflect(engine, schema=schema, only=only)
```

Sorry for not being able to provide more details, but I'm happy to answer any questions and test proposed changes.

**Comment by zzzeek** [comment, marked as answer]
your issue is due to URL changes in #6329 and it's not valid to have a question mark "?" in the query string of your URL, e.g. ``?autocommit=True``.   take a look at https://docs.sqlalchemy.org/en/14/dialects/mssql.html#hostname-connections and you probably mean to have an ampersand there.
